src/main/java/Duke.java

```diff
@@ -1,10 +1,120 @@
-                + "|____/ \\__,_|_|\\_\\___|\n";
+        initializeDuke(args);
+        printUserGreet();
+        runDuke();
+        printUserFarewell();
+    }
+
+    /**
+     * Assign the appellation since this in the only thing to initialize
+     *
+     * @param args the appellation for the guest, should only be assigned to {@value #userName} or None
+     */
+    private static void initializeDuke(String[] args) {
+        if (args.length >= 1) {
+            userName = args[args.length - 1];
+        }
+    }
+
+    /**
+     * Return only when user's command containing word "bye"
+     * Get command from user, parse the command and then execute the corresponding command.
+     */
+    private static void runDuke() {
+        while (true) {
+            printUserPrompt();
+            Command command = CommandManager.getCommand();
+            printLineDivider();
+            if (exitIfContainExitCommand(command)) return;
+            CommandManager.runCommand(command);
+            printLineDivider();
+        }
+    }
+
+    private static boolean exitIfContainExitCommand(Command command) {
+        for (String token : command.getCommandTokens()) {
+            if (isExiting(token)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Print a line Divider, making the layout more clear
+     */
+    private static void printLineDivider() {
+        // TODO Beautify
+        System.out.println("------------------------------");
+    }
+
+    /**
+     * Greet the user, called right after initialization
+     */
+    private static void printUserGreet() {
+        // TODO Personalize the greeting information (maybe change DUKE to BECK or something else)
+        String logo = """
+                 ____        _       \s
+                |  _ \\ _   _| | _____\s
+                | | | | | | | |/ / _ \\
+                | |_| | |_| |   <  __/
+                |____/ \\__,_|_|\\_\\___|
+                """;
         System.out.println("Hello from\n" + logo);
+        printLineDivider();
+        System.out.println("Hello, I'm Duke");
+        System.out.println("What can I do for you? " + userName);
+        printLineDivider();
+    }
+
+    /**
+     * Say goodbye to user, called when program is ready to exit
+     */
+    private static void printUserFarewell() {
+        System.out.println("Bye, " + userName + ". Hope to see you soon!");
+        printLineDivider();
+    }
+
+    /**
+     * Print {@value #userName} and prompt symbol, making the layout more clear
+     */
+    private static void printUserPrompt() {
+        // TODO add more prompt style
+        System.out.print(userName + " > ");
+    }
+
+
+    /**
+     * Check whether the token is contained in exit command
+     *
+     * @param token the command token to be checked
+     * @return whether the token is contained in exit command
+     */
+    private static boolean isExiting(String token) {
```

Comment:
can be named better, eg isTokenInExit

Reply:
> can be named better, eg isTokenInExit
Similar method exitIfContainExitCommand(), thanks for reviewing.